homeassistant/components/feedreader/__init__.py

```diff
@@ -2,61 +2,117 @@
 DEFAULT_SCAN_INTERVAL = timedelta(hours=1)
 
+MY_KEY: HassKey[StoredData] = HassKey(DOMAIN)
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: {
-            vol.Required(CONF_URLS): vol.All(cv.ensure_list, [cv.url]),
-            vol.Optional(
-                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-            ): cv.time_period,
-            vol.Optional(
-                CONF_MAX_ENTRIES, default=DEFAULT_MAX_ENTRIES
-            ): cv.positive_int,
-        }
-    },
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_URLS): vol.All(cv.ensure_list, [cv.url]),
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                    ): cv.time_period,
+                    vol.Optional(
+                        CONF_MAX_ENTRIES, default=DEFAULT_MAX_ENTRIES
+                    ): cv.positive_int,
+                }
+            )
+        },
+    ),
     extra=vol.ALLOW_EXTRA,
 )
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Feedreader component."""
-    urls: list[str] = config[DOMAIN][CONF_URLS]
-    if not urls:
-        return False
-
-    scan_interval: timedelta = config[DOMAIN][CONF_SCAN_INTERVAL]
-    max_entries: int = config[DOMAIN][CONF_MAX_ENTRIES]
     storage = StoredData(hass)
     await storage.async_setup()
-    feeds = [
-        FeedReaderCoordinator(hass, url, scan_interval, max_entries, storage)
-        for url in urls
-    ]
+    hass.data[MY_KEY] = storage
+
+    if DOMAIN in config:
+        for url in config[DOMAIN][CONF_URLS]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_IMPORT},
+                    data={
+                        CONF_URL: url,
+                        CONF_MAX_ENTRIES: config[DOMAIN][CONF_MAX_ENTRIES],
+                    },
+                )
+            )
+
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2025.1.0",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Feedreader",
+            },
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: FeedReaderConfigEntry) -> bool:
+    """Set up Feedreader from a config entry."""
+    storage = hass.data[MY_KEY]
 
-    await asyncio.gather(*[feed.async_refresh() for feed in feeds])
+    coordinator = FeedReaderCoordinator(
+        hass,
+        entry.data[CONF_URL],
+        DEFAULT_SCAN_INTERVAL,
```

Comment:
I'm guessing you didn't move this to the coordinator because for a follow up

Reply:
not intentional :see_no_evil: will remove it from the signature of the coordinator and use it directly